wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/Container/ViewModel/ConversationListViewControllerViewModel+StartUIDelegate.swift

```diff
@@ -23,10 +23,38 @@ import WireSyncEngine
 
 extension ConversationListViewController.ViewModel: StartUIDelegate {
     func startUI(_ startUI: StartUIViewController, didSelect user: UserType) {
-        oneToOneConversationWithUser(user) { result in
-            guard case .success(let conversation) = result else { return }
+        guard let userID = user.qualifiedID else { return }
 
-            ZClientViewController.shared?.select(conversation: conversation, focusOnView: true, animated: true)
+        let conversation = user.oneToOneConversation
+
+        Task {
+            do {
+                let status = try await userSession.oneOnOneConversationCreationStatus.invoke(userID: userID)
+
+                switch status {
+                case .exists(protocol: .proteus, established: _),
+                        .exists(protocol: .mls, established: true):
+
+                    // If the conversation exists, and is established (in case of mls),
+                    // then we open the conversation
+                    guard let conversation else { return }
+                    await openConversation(conversation)
+
+                case .exists(protocol: .mls, established: false),
+                        .doesNotExist(protocol: _):
+
+                    // If the conversation should be using mls and is not established,
+                    // or does not exits, then we open the user profile to let the user
+                    // create the conversation
+                    await openUserProfile(user)
+
+                default:
+                    break
+                }
+
+            } catch {
+
```

Comment:
Is this intentionally empty?

Reply:
I meant to log the error. nice catch https://github.com/wireapp/wire-ios/pull/1321/commits/5bf1d89609b0ea0c3c77278eb4fd6aacc0331a89